src/logged/mod.rs

```diff
@@ -155,17 +155,16 @@ async fn find_offset(
                 } else {
                     StorableRecordDecoder::decode
                 };
-                match decode_fn(&mut decoder, &mut buf) {
-                    Ok(Some(record)) if beginning_offset.is_none() => {
+                match decode_fn(&mut decoder, &mut buf)? {
+                    Some(record) if beginning_offset.is_none() => {
                         beginning_offset = Some(record.offset);
                         end_offset = Some(record.offset);
                     }
-                    Ok(Some(record)) => {
+                    Some(record) => {
                         end_offset = Some(record.offset);
                     }
-                    Ok(None) if len == 0 => break,
-                    Ok(None) => (),
-                    Err(e) => return Err(e),
+                    None if len == 0 => break,
+                    None => (),
```

Comment:
Using the error propagation operator here simplifies this block without changing the behaviour. I'm not sure you're using this anywhere else though so if there's a conscious choice to avoid using that then I can revert this bit of course :)

Reply:
Nice!